tests/state/state.go

```diff
@@ -0,0 +1,301 @@
+// Copyright 2025 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package state
+
+import (
+	"context"
+	"testing"
+
+	"github.com/ipfs/go-cid"
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/onsi/gomega/types"
+
+	"github.com/sourcenetwork/immutable"
+
+	acpIdentity "github.com/sourcenetwork/defradb/acp/identity"
+	"github.com/sourcenetwork/defradb/client"
+	"github.com/sourcenetwork/defradb/crypto"
+	"github.com/sourcenetwork/defradb/event"
+	netConfig "github.com/sourcenetwork/defradb/net/config"
+	"github.com/sourcenetwork/defradb/node"
+	"github.com/sourcenetwork/defradb/tests/clients"
+)
+
+// StatefulMatcher is a matcher that requires state to be reset between tests.
+type StatefulMatcher interface {
+	types.GomegaMatcher
+	// ResetMatcherState resets the state of the matcher.
+	ResetMatcherState()
+}
+
+type DatabaseType string
+
+// KMSType is the type of KMS to use.
+type KMSType string
+
+type ClientType string
+
+type ColDocIndex struct {
+	Col int
+	Doc int
+}
+
+func NewColDocIndex(col, doc int) ColDocIndex {
+	return ColDocIndex{col, doc}
+}
+
+// P2pState contains all p2p related testing state.
+type P2pState struct {
```

Comment:
suggestion: `P2PState`.

I'll add it here only but the reference for this suggestion and the others to follow: https://go.dev/wiki/CodeReviewComments#initialisms